### ISSUE TITLE
curl: (60) SSL certificate problem using pyenv install

### ISSUE THREAD

**Issue by Pavel-king** [pyenv/pyenv#3016]
Too many issues will kill our team's development velocity, drastically.
Make sure you have checked all steps below.

### Prerequisite
* [x] Make sure your problem is not listed in [the common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems).
* [x] Make sure no duplicated issue has already been reported in [the pyenv issues](https://github.com/pyenv/pyenv/issues). You should look for closed issues, too.
* [x] Make sure you are not asking us to help solving your specific issue.
  * GitHub issues is opened mainly for development purposes. If you want to ask someone to help solving your problem, go to some community site like [Gitter](https://gitter.im/yyuu/pyenv), [StackOverflow](https://stackoverflow.com/questions/tagged/pyenv), etc.
* [x] Make sure your problem is not derived from packaging (e.g. [Homebrew](https://brew.sh)).
  * Please refer to the package documentation for the installation issues, etc.
* [x] Make sure your problem is not derived from plugins.
  * This repository is maintaining `pyenv` and the default `python-build` plugin only. Please refrain from reporting issues of other plugins here.

### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): macOS Sonoma, Version 14.5
- [x] OS architecture (e.g. amd64): Apple M1 Max
- [x] pyenv version: pyenv 2.4.7
- [x] Python version: try to install 3.12.4
- [x] C Compiler information (e.g. gcc 7.3): 
          Apple clang version 15.0.0 (clang-1500.3.9.4)
          Target: arm64-apple-darwin23.5.0
          Thread model: posix
           InstalledDir: /Library/Developer/CommandLineTools/usr/bin
- [ ] Please attach the debug trace of the failing command as a gist:
  * Run `env PYENV_DEBUG=1 <faulty command> 2>&1 | tee trace.log` and attach `trace.log`. E.g. if you have a problem with installing Python, run `env PYENV_DEBUG=1 pyenv install -v <version> 2>&1 | tee trace.log` (note the `-v` option to `pyenv install`).
- [ ] If you have a problem with installing Python, please also attach `config.log` from the build directory
  * The build directory is reported after the "BUILD FAILED" message and is usually under `/tmp`.
- [ ] If the build succeeds but the problem is still with the build process (e.g. the resulting Python is missing a feature), please attach
  * the debug trace from reinstalling the faulty version with `env PYENV_DEBUG=1 pyenv install -f -k -v <version> 2>&1 | tee trace.log`
  * `config.log` from the build directory. When using `pyenv install` with `-k` as per above, the build directory will be under `$PYENV_ROOT/sources`.

**Comment by Pavel-king** [comment]
when I try use pyenv install 3.12.4, it says some SSL error. I can use `curl -k https://www.python.org/ftp/python/3.12.4/Python-3.12.4.tar.xz` to disable SSL and get the file. But I dont know how to disable SSL when I use pyenv install, is there an option ?


`python-build: use openssl@3 from homebrew
python-build: use readline from homebrew
Downloading Python-3.12.4.tar.xz...
-> https://www.python.org/ftp/python/3.12.4/Python-3.12.4.tar.xz
error: failed to download Python-3.12.4.tar.xz

BUILD FAILED (OS X 14.5 using python-build 20180424)

Results logged to /var/folders/td/gqj3t0m53h3_8wwjyp43vjsh0000gs/T/python-build.20240713135800.62986.log

Last 10 log lines:
/var/folders/td/gqj3t0m53h3_8wwjyp43vjsh0000gs/T/python-build.20240713135800.62986 ~/tmp
warning: xz not found; consider installing `xz` package
curl: (60) SSL certificate problem: unsupported or invalid name syntax
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
`

**Comment by native-api** [comment]
```
warning: xz not found; consider installing `xz` package
curl: (60) SSL certificate problem: unsupported or invalid name syntax
More details here: https://curl.se/docs/sslcerts.html
```

This however should not happen under MacOS. Where is your curl installed? Did you tamper with `PATH`? Did you install the Python build prerequisites?

Please provide the debug logs as per the issue template for us to be able to see what's happening.

**Comment by Pavel-king** [comment]
Thanks, @native-api

the curl issue fixed when I add below env variable
`export PYTHON_BUILD_CURL_OPTS="-k"`
And there is a warning when I install 3.12.4, 
`WARNING: The Python lzma extension was not compiled. Missing the lzma lib?`
It got fixed when I install lzma module
`brew install readline xz`

I'm not sure from where I installed curl since it's a long time. Maybe it's built in system or from some external script.
I have installed 2.7.18 & 3.10.0 before. I think python build prerequisites are installed
